x/stake/cross_stake/cross_stake.go

```diff
@@ -0,0 +1,307 @@
+package cross_stake
+
+import (
+	"strconv"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/bsc"
+	"github.com/cosmos/cosmos-sdk/bsc/rlp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/stake/types"
+)
+
+type CrossStakeApp struct {
+	stakeKeeper Keeper
+}
+
+func NewCrossStakeApp(stakeKeeper Keeper) *CrossStakeApp {
+	return &CrossStakeApp{
+		stakeKeeper: stakeKeeper,
+	}
+}
+
+func (app *CrossStakeApp) ExecuteAckPackage(ctx sdk.Context, payload []byte) sdk.ExecuteResult {
+	app.stakeKeeper.Logger(ctx).Error("received cross stake ack package ")
+	return sdk.ExecuteResult{}
+}
+
+func (app *CrossStakeApp) ExecuteFailAckPackage(ctx sdk.Context, payload []byte) sdk.ExecuteResult {
+	app.stakeKeeper.Logger(ctx).Error("received cross stake fail ack package ")
+	return sdk.ExecuteResult{}
+}
+
+func (app *CrossStakeApp) ExecuteSynPackage(ctx sdk.Context, payload []byte, relayFee int64) sdk.ExecuteResult {
+	eventType, err := DeserializeCrossStakeSynPackage(payload)
+	if err != nil {
+		app.stakeKeeper.Logger(ctx).Error("unmarshal cross stake sync claim error", "err", err.Error(), "claim", string(payload))
+		panic("unmarshal cross stake claim error")
+	}
+
+	var result sdk.ExecuteResult
+	switch eventType {
+	case types.CrossStakeTypeDelegate:
+		result, err = app.handleDelegate(ctx, payload, relayFee)
+	case types.CrossStakeTypeUndelegate:
+		result, err = app.handleUndelegate(ctx, payload, relayFee)
+	case types.CrossStakeTypeRedelegate:
+		result, err = app.handleRedelegate(ctx, payload, relayFee)
+	}
+	if err != nil {
+		panic(err)
+	}
+
+	return result
+}
+
+func (app *CrossStakeApp) handleDelegate(ctx sdk.Context, payload []byte, relayFee int64) (sdk.ExecuteResult, error) {
+	var pack types.CrossStakeDelegateSynPackage
+	if err := rlp.DecodeBytes(payload, &pack); err != nil {
+		app.stakeKeeper.Logger(ctx).Error("unmarshal cross stake delegate sync claim error", "err", err.Error(), "claim", string(payload))
+		return sdk.ExecuteResult{}, err
+	}
+
+	sideChainId := app.stakeKeeper.ScKeeper.BscSideChainId(ctx)
+	if scCtx, err := app.stakeKeeper.ScKeeper.PrepareCtxForSideChain(ctx, sideChainId); err != nil {
+		return sdk.ExecuteResult{}, err
+	} else {
+		ctx = scCtx
+	}
+
+	delAddr, err := types.GetStakeCAoB(pack.DelAddr[:], "Delegate")
```

Comment:
Can we simplify it?  like `sdk.XOR(tmhash.SumTruncated([]byte(sideChainId)), operator)`

Reply:
Done